src/site/topic20.rst

```diff
@@ -93,6 +93,47 @@ Recursive
 Binary Search
 =============
 
+* Let's say you're looking for page 554 in a textbook
+* You'd probably open the book somewhere in the middle-ish
+* If the page you end up turning to is page 383, which half of the book should you continue your search on?
+    * Left pages, or right pages?
+* Since 554 is greater than 383, it must be in the set of right pages
+    * We eliminate the 383 pages from the set of left pages from our search since 554 must **not** be there
+* Next, you turn to a page somewhere between pages 383 and the end of the book and see page number 621
+    * Which set of pages do you then continue your search down?
+
+* This general idea is the basis of binary search
```

Comment:
Technically, that's [interpolation search](https://github.com/jameshughes89/cs102/blame/f8745ea4ced7b862c74187c6a23fedd390195609/src/site/topic20.rst#L15-L16), but I won't split hairs here. The section header is already binary search, so it'd be a little odd to be like "and that, my friends, is interpolation search :microphone: :wave: "